67 auth

myc37 opened the pull request:
- [x] remove firebase auth
- [x] migrate to express/postgres auth
- [ ] implement jwt auth

myc37 committed:
chore: abstract useAuth types

myc37 committed:
chore: abstract useAuth types

myc37 committed:
feat: migrate from firebase auth to postgres/express

myc37 committed:
feat: migrate from firebase auth to postgres/express

myc37 committed:
feat: add jwt auth

myc37 committed:
feat: add jwt auth

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -1,47 +1,15 @@
 
 const AuthContext = createContext<IAuth>({
   user: null,
+  loading: false,
```
```suggestion
  isLoading: false,
```

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -63,15 +31,29 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
   const [loading, setLoading] = useState<boolean>(true);
```
```suggestion
  const [isLoading, setLoading] = useState(true);
```

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -63,15 +31,29 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
-  );
+  useEffect(() => {
+    const checkUserAuthorization = async (email: string) => {
+      const token = localStorage.getItem("jwt_token");
```
eh I thought we using cookies not localStorage?

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -63,15 +31,29 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
+      setUser(user as User);
+    };
+
+    if (user) {
```
user will be initiated as null right, so how will the checkUserAuthorization function ever be called?

raynerljm replied:
we discuss tomorrow on we should do the auth

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -111,62 +96,64 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
+    setLoading(true);
+    const apiServiceBuilder = new ApiServiceBuilder({
+      method: HTTP_METHOD.POST,
+      endpoint: `/${email}`,
```
```suggestion
      endpoint: `/users/${email}`,
```

raynerljm commented on hooks/useAuth/useAuth.types.ts:
```diff
@@ -0,0 +1,39 @@
+  password: string;
+};
+
+export type User = {
```
there's already a User type in users.ts, should either use the same or consolidate somehow

myc37 committed:
Merge branch '67-auth' of https://github.com/orbital-skylab/skylab-frontend into 67-auth

myc37 committed:
chore: move useAuth types to types directory

myc37 committed:
feat: update useAuth to cookie authentication

myc37 committed:
feat: update auth context

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -59,27 +37,56 @@ interface AuthProviderProps {
+  const [user, setUser] = useState<User>(defaultUser);
   const [loading, setLoading] = useState<boolean>(true);
+  const [currentCohortYear, setCurrentCohortYear] =
+    useState<number>(defaultCohortYear);
```
```suggestion
    useState<Cohort["academicYear"]>(defaultCohortYear);
```
can make reference to the cohort academic year type (which is still number)

myc37 committed:
feat: update auth context and add cohort context

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -54,32 +23,39 @@ const AuthContext = createContext<IAuth>({
 export const AuthProvider = ({ children }: AuthProviderProps) => {
   const [user, setUser] = useState<User | null>(null);
-  const [loading, setLoading] = useState<boolean>(true);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```
no need to add boolean, the typing is alrdy implicit

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -54,32 +23,39 @@ const AuthContext = createContext<IAuth>({
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [auth]
-  );
+  useEffect(() => {
```
actually i think you could replace this whole chunk with useFetch

raynerljm commented on hooks/useAuth/useAuth.tsx:
```diff
@@ -89,91 +65,143 @@ export const AuthProvider = ({ children }: AuthProviderProps) => {
 
   const signIn = async (email: string, password: string) => {
-    await signInWithEmailAndPassword(auth, email, password);
+    setIsLoading(true);
```
i think this is okay but we need to watch if this causes issues
(eg. signing in sets loading to true so if the landing page has any authIsLoading checks, it would cause the whole page to change and flicker)

raynerljm commented on hooks/useCohort/useCohort.tsx:
```diff
@@ -0,0 +1,58 @@
+
+const CohortContext = createContext<ICohort>({
+  currentCohortYear: DEFAULT_COHORT_YEAR,
+  loading: false,
```
standardize loading to isLoading

raynerljm commented on hooks/useCohort/useCohort.tsx:
```diff
@@ -0,0 +1,58 @@
+
+export const CohortProvider = ({ children }: CohortProviderProps) => {
+  const [currentCohortYear, setCurrentCohortYear] =
+    useState<number>(DEFAULT_COHORT_YEAR);
```
```suggestion
    useState<Cohort["academicYear"]>(DEFAULT_COHORT_YEAR);
```

raynerljm commented on hooks/useCohort/useCohort.tsx:
```diff
@@ -0,0 +1,58 @@
+export const CohortProvider = ({ children }: CohortProviderProps) => {
+  const [currentCohortYear, setCurrentCohortYear] =
+    useState<number>(DEFAULT_COHORT_YEAR);
+  const [loading, setLoading] = useState<boolean>(true);
```
```suggestion
  const [loading, setLoading] = useState(false);
```
default should be false because the useEffect sets loading as true

raynerljm commented on hooks/useCohort/useCohort.tsx:
```diff
@@ -0,0 +1,58 @@
+} from "@/types/useCohort";
+
+const CohortContext = createContext<ICohort>({
+  currentCohortYear: DEFAULT_COHORT_YEAR,
```
default should be null because what if the year formats are changed by admins to '22' instead of '2022', then the default would be wrong

raynerljm commented on types/useCohort.ts:
```diff
@@ -0,0 +1,10 @@
+export interface ICohort {
+  currentCohortYear: number;
+  loading: boolean;
```
```suggestion
 isLoading: boolean;
```

raynerljm commented on types/users.ts:
```diff
@@ -9,7 +7,6 @@ export interface User {
   linkedinUrl?: string;
   personalSiteUrl?: string;
   selfIntro?: string;
```
you should rebase/merge main first because of the type overhaul i pushed a few days ago

raynerljm reviewed (changes requested):
mostly looks good, do rebase from main and fix comments and it shd be good to go

myc37 committed:
fix: loading variable names and useState types

myc37 committed:
fix: resolve merge conflicts except casing issue

myc37 committed:
fix: rename placeholder

myc37 committed:
fix: rename FormControllers to formControllers

raynerljm commented on hooks/useCohort/useCohort.tsx:
```diff
@@ -0,0 +1,56 @@
+});
+
+export const CohortProvider = ({ children }: CohortProviderProps) => {
+  const [currentCohortYear, setCurrentCohortYear] = useState<
+    Cohort["academicYear"] | null
+  >(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const getCurrentCohortYear = async () => {
+      setIsLoading(true);
+      const apiServiceBuilder = new ApiServiceBuilder({
+        method: HTTP_METHOD.GET,
+        endpoint: "/cohorts/latest",
+      });
+      const apiService = apiServiceBuilder.build();
+      const latestCohortYearResponse = await apiService();
+
+      if (latestCohortYearResponse.ok) {
+        const { academicYear } = await latestCohortYearResponse.json();
+        setCurrentCohortYear(Number(academicYear));
+      }
+      setIsLoading(false);
+    };
+
+    getCurrentCohortYear();
+  }, []);
```
```suggestion
const { data: currentCohort, status: currentCohortYearStatus } = useFetch<Cohort>({ endpoint: "/cohorts/latest"});
const { data: cohorts, status: cohortsStatus } = useFetch<Cohort[]>({ endpoint: "/cohorts"});

const currentCohortYear = currentCohort && currentCohort.academicYear;

const isLoading = isFetching(currentCohortYearStatus, cohortsStatus);

```

raynerljm committed:
refactor: make projects and staff page use the new cohort context

Note that they are untested so please do testing. Additionally, the milestones page will be modified to use the new cohort context in the PR 64-milestones

myc37 committed:
feat: update useCohort to use useFetch

myc37 committed:
fix: format with prettier

raynerljm committed:
feat: fix context children passing

raynerljm committed:
Merge branch 'staging' into 67-auth

raynerljm committed:
Merge branch 'staging' into 67-auth

myc37 committed:
feat: remove token logic from api helpers

myc37 committed:
feat: update useAuth and add proxy to dev api

myc37 committed:
feat: wrap Hero page with Body component

myc37 committed:
feat: update useCohort children passing and useEffect dependency array

myc37 committed:
Merge branch '67-auth' of https://github.com/orbital-skylab/skylab-frontend into 67-auth

myc37 committed:
feat: update useAuth initial loading state to false

raynerljm committed:
fix: fix slicing of roles

raynerljm committed:
refactor: change to include credentials

raynerljm committed:
fix: fix hero padding
